### PR TITLE
Repository pattern implementation

### PR DESCRIPTION
**Pull request by based-kwl**
- Repository pattern implemented for all existing routes
- Model schema changes
- Code cleanup

model schemas changes:
- removed `messages` attribute from `StudyRoom` - retrieve the messages using the available routes.
- `StudyRooms` -> `studyRooms` in `Student` schema.
- `sID` -> `studyRoomID` in `StudyRoom` schema and everywhere else that uses it
- `mID` -> `messageID` in `Message` schema
- `cnID` -> `courseNotesID` in `CourseNotes` schema
- `cID` -> `calendarID` in `Calendar` schema
- added `eventID` to `Event` schema

TODO:
validation

### CODE REVIEW

**Comment by guaj** [comment]
need to create a branch in front end to apply the changes in the var namings. We cant approve this request as long as its not breaking the functionnality of our deployed app .

**Comment by mnashat** [comment]
> need to create a branch in front end to apply the changes in the var namings. We cant approve this request as long as its breaking the functionnality of our deployed app . Else good work on the code. Code is clean and well documented.

Agreed. Very well documented and structured. Great job! And as mentioned, we can't merge this yet as this will break compatibility with frontend live deployment. We'll refactor frontend and merge both. Well done!

**Comment by mnashat** [comment]
It is extremely important that new backend branches be created from this branch so that new frontend and backend implementations are consistent from this point onwards (until the current frontend is updated for compatibility). We don't want newer branches implementing outdated routes/code structure/branches, as this will be a huge pain to fix later on. Please take note everyone!

@based-kwl @guaj @RamB98 @EyalAzimov @jacobmathewlopez

**Comment by guaj** [comment]
> It is extremely important that new backend branches be created from this branch so that new frontend and backend implementations are consistent from this point onwards (until the current frontend is updated for compatibility). We don't want newer branches implementing outdated routes/code structure/branches, as this will be a huge pain to fix later on. Please take note everyone!
> 
> @based-kwl @guaj @RamB98 @EyalAzimov @jacobmathewlopez

This branch needs to be merged before we start from this branch. Else we will push things that are not actually working. I recommend we do this beginning of next sprint.

**Comment by mnashat** [comment]
> > It is extremely important that new backend branches be created from this branch so that new frontend and backend implementations are consistent from this point onwards (until the current frontend is updated for compatibility). We don't want newer branches implementing outdated routes/code structure/branches, as this will be a huge pain to fix later on. Please take note everyone!
> > @based-kwl @guaj @RamB98 @EyalAzimov @jacobmathewlopez
> 
> This branch needs to be merged before we start from this branch. Else we will push things that are not actually working. I recommend we do this beginning of next sprint.

Noted!

**Comment by mnashat** [comment]
> > It is extremely important that new backend branches be created from this branch so that new frontend and backend implementations are consistent from this point onwards (until the current frontend is updated for compatibility). We don't want newer branches implementing outdated routes/code structure/branches, as this will be a huge pain to fix later on. Please take note everyone!
> > @based-kwl @guaj @RamB98 @EyalAzimov @jacobmathewlopez
> 
> This branch needs to be merged before we start from this branch. Else we will push things that are not actually working. I recommend we do this beginning of next sprint.

Agreed, but it looks like some things are already broken due to schema changes on the database, so we might need to update everything ASAP anyways.

File: src/routes/room.js
Comment (mnashat) [inline src/routes/room.js:12]: Add `const data = req.body;` here. Variable `data` is not defined, so if conditions aren't executing properly.

File: src/routes/room.js
Comment (mnashat) [inline src/routes/room.js:32]: Method `updateOne(...)` does not exist in `StudyRoomRepository`. For updating a room, the method is called 'save(...)`. Either change this line to `StudyRoomRepository.save(...)` or rename the method in `StudyRoomRepository` to `updateOne(...)`.

File: src/repository/studyRoomRepository.js
Comment (mnashat) [inline src/repository/studyRoomRepository.js:14]: Method `save(...)` does not exist in `createStudyRoom`.

File: src/repository/studyRoomRepository.js
Comment (mnashat) [inline src/repository/studyRoomRepository.js:41]: Some promise returns such as this one seems to be invalid. Please recheck your code to make sure all promises are returned properly. It is also causing a lot of calls (such as those in `room.js`) to not correctly recognize `.then(...)` calls, which leads to backend `console.log(...)` calls to not print anything in the backend console.

**Review by mnashat — CHANGES_REQUESTED** [review]
There are a lot of issues in this refactor such as invalid calls and returns. Some of the routes are also not implemented correctly. Please check my comments or some examples of these. Please review all changes and correct the code appropriately. Thanks!

**Review by mnashat — APPROVED** [review]
Great job on the refactoring! Approved.

**Review by guaj — APPROVED** [review]

**Pull request merged**
